mkdocs_include_markdown_plugin/event.py

```diff
@@ -54,11 +54,21 @@ def found_include_tag(match):
 
         return text_to_include
 
+    def interpret_escapes(value: str) -> str:
```

Comment:
Since this function is not needed within the scope, could you move it to the beggining of the module?

Reply:
OK, let's see if I follow. This function, interpret_escapes, is only used in the function found_include_markdown_tag, which in turn is local inside the module-level function _on_page_markdown. That means the tightest scope one could put interpret_escapes in would be inside found_include_markdown_tag. I thought about that, but thought typographically it would disrupt the flow of found_include_markdown_tag too much. On the other hand, since it really is just a little abbreviation for a couple of verbose calls to string member functions, I thought it would be good to keep it (a) fairly tightly scoped, and (b) fairly close to its usage point. So that's the rationale for its current location.

Of course I can move it to the module level, i.e. a wider scope, and I can put it at the top of the file (before or after the regexes?) But that makes it visible to a greater portion of the module, where it is not relevant, and puts it quite far from its use point, even though it has the feel of a local throwaway abbreviation. Please confirm that's what you prefer, and whether you'd like it before or after the regexes defined near the top, and I'll be happy to move it if you feel that's an improvement to the code. Thanks!

Reply:
After the regexes or event better would be in the `process.py` module.
It's because inner functions are created every time that the outer function is executed, so to maintain readability the solution is to keep a reasonable set of modules clearly differentiated, not by scoping the functions.

Reply:
OK, I will move it to process.py and push a new version shortly. I will just take a moment to say here that I respectfully disagree with your analysis, but this is your module and of course it should be designed as you see fit. For whatever it's worth, here is my counterpoint: 

The only thing that happens on each call to _on_page_markdown is that that closure of interpret_escapes is updated; but as that local function references no variables other than its parameter, it has empty closure, so this step may quite possibly be compiled away entirely, and in any case is trivial compared to the cost of doing all of the regexp searches and substitutions. Moreover, if one is worried about this epsilon of optimization, then why at the top of _on_page_markdown don't we let page_src_path be Path(page.file.abs_src_path) since that's the only expression which uses page_src_path (three times below) and save all those Path constructor calls? You could say that presumably python performs common subexpression elimination in compiling; but then we may as well presume that it optimizes away the updating of empty closures, too.

Reply:
> Moreover, if one is worried about this epsilon of optimization, then why at the top of _on_page_markdown don't we let page_src_path be Path(page.file.abs_src_path) since that's the only expression which uses page_src_path (three times below) and save all those Path constructor calls?

You are totally right. Thanks for the suggestion, I'll update it before the release :+1:

The point of move this function out not is optimize the code now, it's again a sanity refactor to make that, in the future, another programmer want not to include more and more functions here, which improves performance in case that someone needs to run a great amount of includes in their project.

 > but this is your module and of course it should be designed as you see fit

In fact, this is not my module. You can see in the metadata that I am not the author, I just maintain it, so it's my job that the package doesn't start to get chaotic. If you have suggestion that becomes it more maintainable and faster (always balancing both reasonably), will be welcome like the previous one. Note that I could accept this pull as is and do the changes before the release, so I would ask that, if you have objections regarding the suggestions that I leave you, share them and, if they are reasonable, they will be taken into account, but it does not seem fair to say that you do not agree on how the module is maintained by taking in consideration other parts that are not being discussed here. The fact that are parts unoptimized does not rejects the point that this thread discusses, which is ¿should this function be here? Not, because does not increments readability (although this being subjective), could affect performance and opens the gate of future addition of inner functions. If you don't agree with this, just settle the conversation and I'll do the change myself before launch.

Reply:
As I said I would, I already moved the function to process.py as requested. Thanks, Glen

Reply:
Thank you :+1: